openedx_events/content_authoring/signals.py

```diff
@@ -21,3 +21,39 @@
         "catalog_info": CourseCatalogData,
     }
 )
+
+
+# .. event_type: org.openedx.content_authoring.xblock.published.v1
+# .. event_name: XBLOCK_PUBLISHED
+# .. event_description: Fired when an XBlock is published.
```

Comment:
Quick note: Please be very specific in the description as to exactly when this is fired. For instance:

1. Does it only fire for XBlocks that are changing in the `published` branch, or will it also fire for changes to XBlocks in the draft branch?
2. Does it fire only for when something is initially created, or also on updates?
3. Does it fire for changes that happen to content in content libraries?
4. Does it fire when a parent XBlock has changed, potentially changing the value of inherited fields?
5. Does it fire when a child XBlock has changed (e.g. when it fires for a problem, does it also fire for the Unit it's enclosed in)?

I honestly don't want to engage in a long discussion about when it _should_ fire. I'm guessing you already have some part of the code that you're plugging into that will address your use cases, and I don't want to complicate your implementation. But please be as precise as you can in both the naming and description of these signals.

Reply:
(@navinkarkera: Please take this as a non-binding comment. You already have approval from @felipemontoya, who is a maintainer of this repo.)

Reply:
> If a unit is published, this event will be fired with only its own usage_key, no additional events are published for child or parent blocks.

I see. So if publishing happens at the section or subsection level, there is likewise no event fired for the unit? This would be good to document in the description so people aren't caught by surprise.

Reply:
@ormsbee So if a subsection is published, an event is fired with `block_type=sequential` and if a section is published, event with `block_type=chapter` is fired. Should I include something like:

If a parent block with changes in one or more child blocks is published, only a single `XBLOCK_PUBLISHED` event is fired with parent block details. For example: If a section is published with changes in multiple units, only a single event is fired with section details like : `XBlockData(usage_key="section-usage-key", block_type="chapter")`

Reply:
Yeah, I think that would be a really helpful comment to have in that annotation. Thank you!